cmdstanpy/model.py

```diff
@@ -418,6 +418,35 @@ def code(self) -> Optional[str]:
             )
         return code
 
+    @staticmethod
+    def _is_out_of_date(stan_file: str, exe_time: float,
+                        include_paths: Iterable[str]) -> bool:
+        """
+        Determine whether the Stan program file or any of its includes are
+        newer than the executable.
+        """
+        if os.path.getmtime(stan_file) > exe_time:
+            return True
+
+        # Recursively identify whether any included files changed.
+        pattern = re.compile(r"^\s*#include\s+(.*?)(?:$|\/\/)", re.MULTILINE)
```

Comment:
Not sure if you're looking for feedback at this stage, but regex is notoriously bad at finding `#include`s and this has lead to a lot of bugs in RStan where they use them.

For example,
```stan
functions {
  /*
  #include to load in common functions
  */
  #include funs.stan
}
```

Is something someone might reasonably write but will be almost impossible to disentangle with regex.

There is another option, though, which is use `self.src_info()`. The dictionary returned by that will have a full list of includes, including recursive includes, precisely as the compiler itself interprets them

Reply:
Much nicer! Updated.